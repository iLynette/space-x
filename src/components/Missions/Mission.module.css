table,
th,
td {
  border: 1px solid #dce1e5;
  padding: 1rem;
}

table {
  border-collapse: collapse;
  width: 85%;
  margin: 1rem auto;
}

th {
  text-align: start;
}

tr:nth-child(even) {
  background-color: var(--neutral-color);
}

.mission-name {
  vertical-align: top;
  width: fit-content;
  font-weight: bold;
}

.mission-description {
  width: max-content;
}

.mission-status {
  white-space: nowrap;
}

.not-reserved {
  background-color: var(--neutral-color-2);
  color: var(--white);
  padding: 0.2rem 0.4rem;
  border-radius: 0.7rem;
  font-size: 0.8rem;
}

.mission-not-joined {
  white-space: nowrap;
  background-color: transparent;
  outline: none;
  border: 1px solid var(--black);
  padding: 0.5rem 0.8rem;
  border-radius: 0.3rem;
  cursor: pointer;
}

.mission-joined {
  white-space: nowrap;
  background-color: transparent;
  outline: none;
  border: 1px solid var(--red);
  color: var(--red);
  padding: 0.5rem 0.8rem;
  border-radius: 0.3rem;
  cursor: pointer;
}
