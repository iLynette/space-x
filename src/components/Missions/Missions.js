import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import fetchMissions, { joinMission } from '../../redux/missions/missions';
import styles from './Mission.module.css';

let fetched = false;

const Missions = () => {
  const { mission } = useSelector((state) => state.missionReducer);
  const dispatch = useDispatch();

  useEffect(() => {
    if (!fetched) {
      dispatch(fetchMissions());
      fetched = true;
    }
  }, []);
  if (!fetched) return <h1>I am loading</h1>;

  const handleClick = (id) => {
    dispatch(joinMission(id));
  };

  return (
    <table>
      <tbody>
        <tr>
          <th>Mission</th>
          <th>Description</th>
          <th>Status</th>
          <th> </th>
        </tr>
        {mission.map(
          ({
            mission_name: missionName,
            mission_id: missionId,
            description,
            reserved,
          }) => (
            <tr key={missionId}>
              <td className={styles['mission-name']}>{missionName}</td>
              <td className={styles['mission-description']}>{description}</td>
              <td className={styles['mission-status']}>
                {' '}
                <p className={styles['not-reserved']}>NOT A MEMBER</p>
              </td>
              <td>
                <button
                  id={missionId}
                  onClick={(e) => handleClick(e.target.id)}
                  className={
                    reserved
                      ? styles['mission-joined']
                      : styles['mission-not-joined']
                  }
                  type="button"
                >
                  {reserved ? 'Leave Mission' : 'Join Mission'}
                  {' '}
                  Join Mission
                </button>
              </td>
            </tr>
          ),
        )}
      </tbody>
    </table>
  );
};

export default Missions;
