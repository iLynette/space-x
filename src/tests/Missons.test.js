import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/configureStore';
import Missions from '../components/Missions/Missions';

describe('Missions tests', () => {
  it('render test', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );

    expect(tree).toMatchSnapshot();
  });
});
