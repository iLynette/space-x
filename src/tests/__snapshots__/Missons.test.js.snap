// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions tests render test 1`] = `
<table>
  <tbody>
    <tr>
      <th>
        Mission
      </th>
      <th>
        Description
      </th>
      <th>
        Status
      </th>
      <th>
         
      </th>
    </tr>
  </tbody>
</table>
`;
